### PR TITLE
Niko/user table update

### PR DESCRIPTION
**Pull request by niko79542**
Hey - The user table at its bare minimum have non-null requirements on ID and the name field.

ID is the randomly generated string,
name is the steemit username

### CODE REVIEW

File: src/models/sequelize/User.js [hunk 1]
```diff
@@ -10,11 +10,49 @@ module.exports = (sequelize, DataTypes) => {
...
+        allowNull: false,
+        unique: true,
+        index: true,
+      },
       createdAt: {
         type: DataTypes.DATE,
         defaultValue: DataTypes.NOW,
-        field: 'created_at'
-      }
+        field: 'created_at',
```
Comment (jay-ithiel) [inline src/models/sequelize/User.js]: we are using `camelCase` everywhere else, but `snake_case` here. Can we keep it consistent?

File: src/services/postService.js [hunk 2]
```diff
@@ -41,7 +43,7 @@ module.exports = {
 
   getPostsOfAuthors: function(authors) {
     const authorObjs = authors.map(name => {
-      return { id: name };
+      return { name };
```
Comment (jay-ithiel) [inline src/services/postService.js]: 2 things here

1) Could use an implicit return here
```
const authorObjs = authors.map(name => { name });
```

2) it's not good to use types in variable names. `authorObjs` could be renamed `authorNames`, or `authors`
```
const authorNames = authors.map(name => { name });
```

File: src/services/replyService.js [hunk 3]
```diff
@@ -21,9 +22,12 @@ class ReplyService {
...
     return PostModel.findById(this.postId) //find post
       .then(postRecord => {
-        // store post info...
-        this.postAuthor = postRecord.userId;
         this.postPermLink = postRecord.permLink;
+        return UserModel.findById(postRecord.userId);
+      })
+      .then(userRecord => {
+        // store post info...
+        this.postAuthor = userRecord.name;
```
Comment (jay-ithiel) [inline src/services/replyService.js]: Is there a reason we have to set `postAuthor` as a class variable? This is a service so there really shouldn't be anything attached to the service

File: src/services/replyService.js [hunk 4]
```diff
@@ -39,22 +43,23 @@ class ReplyService {
       params,
       parentId
     } = this.determineParamOptions(options);
-
+    console.log("the params are", params);
```
Comment (jay-ithiel) [inline src/services/replyService.js]: remove console log. also, double quotes `""` are reserved for html. javascript should be single quotes `''`

File: src/models/sequelize/User.js (hunk 1 above)
Comment (niko79542) [reply src/models/sequelize/User.js]: EDIT:  Nvm you are right.  I will change to camel.

File: src/services/replyService.js (hunk 3 above)
Comment (niko79542) [reply src/services/replyService.js]: I believe this was fixd in the previous pr =)

**Pull request merged**